### ISSUE TITLE
[BUG] Number of device shots in `qml.sample` not considered in tape mode

### ISSUE THREAD

**Issue by glassnotes** [PennyLaneAI/pennylane#1036]
#### Issue description

When using `qml.sample` in tape mode, the number of shots does not match the number of shots specified for the device. 

* *Expected behavior:* Number of samples should be number of shots on the device.

* *Actual behavior:* Number of samples is 1000.

* *Reproduces how often:* Always

* *System information:* (post the output of `import pennylane as qml; qml.about()`)

```
Name: PennyLane
Version: 0.14.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: None
Author-email: None
License: Apache License 2.0
Location: /home/olivia/Software/anaconda3/envs/xanadu/lib/python3.8/site-packages
Requires: numpy, scipy, appdirs, networkx, semantic-version, autograd, toml
Required-by: PennyLane-SF, pennylane-qulacs, PennyLane-qsharp, PennyLane-qiskit, PennyLane-Qchem, PennyLane-Forest, PennyLane-Cirq
Platform info:           Linux-5.4.0-64-generic-x86_64-with-glibc2.10
Python version:          3.8.5
Numpy version:           1.19.5
Scipy version:           1.4.1
Installed devices:
- default.gaussian (PennyLane-0.14.0.dev0)
- default.mixed (PennyLane-0.14.0.dev0)
- default.qubit (PennyLane-0.14.0.dev0)
- default.qubit.autograd (PennyLane-0.14.0.dev0)
- default.qubit.jax (PennyLane-0.14.0.dev0)
- default.qubit.tf (PennyLane-0.14.0.dev0)
- default.tensor (PennyLane-0.14.0.dev0)
- default.tensor.tf (PennyLane-0.14.0.dev0)
- strawberryfields.fock (PennyLane-SF-0.11.0)
- strawberryfields.gaussian (PennyLane-SF-0.11.0)
- strawberryfields.remote (PennyLane-SF-0.11.0)
- qulacs.simulator (pennylane-qulacs-0.12.0)
- microsoft.QuantumSimulator (PennyLane-qsharp-0.8.0)
- qiskit.aer (PennyLane-qiskit-0.13.0.dev0)
- qiskit.basicaer (PennyLane-qiskit-0.13.0.dev0)
- qiskit.ibmq (PennyLane-qiskit-0.13.0.dev0)
- forest.numpy_wavefunction (PennyLane-Forest-0.11.0)
- forest.qpu (PennyLane-Forest-0.11.0)
- forest.qvm (PennyLane-Forest-0.11.0)
- forest.wavefunction (PennyLane-Forest-0.11.0)
- cirq.mixedsimulator (PennyLane-Cirq-0.13.0)
- cirq.pasqal (PennyLane-Cirq-0.13.0)
- cirq.qsim (PennyLane-Cirq-0.13.0)
- cirq.qsimh (PennyLane-Cirq-0.13.0)
- cirq.simulator (PennyLane-Cirq-0.13.0)

```

#### Source code and tracebacks

Please include any additional code snippets and error tracebacks related
to the issue here.

```
import pennylane as qml

dev = qml.device("default.qubit", wires=2, shots=10)

@qml.qnode(dev)
def circuit(x):
    qml.RX(x, wires=0)
    qml.Hadamard(wires=1)
    qml.CNOT(wires=[0, 1])
    return qml.sample(qml.PauliY(0))

circuit(0.3)
```
gives
```
array([-1, -1,  1,  1, -1, -1,  1,  1, -1,  1])
```

Adding `qml.enable_tape()` before device creation gives array of length 1000.

**Comment by josh146** [comment]
This is fixed in #1035 🙂 If you could test that branch quickly and see if it fixes the issue, that would be great!

**Comment by glassnotes** [comment]
@josh146 Yes it appears to be fixed with that :+1: (I would never have guessed this from the PR name!). Thanks!
